pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java

```diff
@@ -269,9 +269,11 @@ public ByteBuffer getState(String key) {
 
         if (producer == null) {
             try {
-                Producer<O> newProducer = ((ProducerBuilderImpl<O>) producerBuilder.clone()).schema(schema).create();
+                Producer<O> newProducer = ((ProducerBuilderImpl<O>) producerBuilder.clone())
```

Comment:
can we please add unit-test here which covers this path.

Reply:
Added